apps/web/components/dashboard/preview/ActionBar.tsx

```diff
@@ -100,14 +91,18 @@ export default function ActionBar({ bookmark }: { bookmark: ZBookmark }) {
         </TooltipContent>
       </Tooltip>
       <Tooltip delayDuration={0}>
+        <DeleteBookmarkConfirmationDialog
+          bookmark={bookmark}
+          open={deleteBookmarkDialogOpen}
+          setOpen={setDeleteBookmarkDialogOpen}
+          onPendingChange={setIsPendingDelete}
+        />
         <TooltipTrigger asChild>
           <ActionButton
-            loading={pendingDeletion}
+            loading={isPendingDelete}
             className="size-14 rounded-full bg-background"
             variant="none"
-            onClick={() => {
-              deleteBookmark({ bookmarkId: bookmark.id });
-            }}
+            onClick={() => setDeleteBookmarkDialogOpen(true)}
           >
```

Comment:
Given that this button no longer itself takes an action, we should switch it to just a normal button instead of `ActionButton`. That way, you also won't need to propagate the pending state back to this component.

Reply:
Fixed in [add delete bookmark description to i18n](https://github.com/hoarder-app/hoarder/pull/787/commits/52924c63425199140a4fba762518a2a8c01d1fc4)